Add multihost testing

engelmi opened the pull request:
Fixes #164 

This PR adds support for specifying pipeline settings - including the pipeline type (currently only `multihost`). The API specification of testing farm for requesting new test run is used as basis:
https://api.testing-farm.io/redoc#operation/request_a_new_test_v0_1_requests_post
(Some inline comments to the old API spec have been updated as well)

engelmi commented:
The code coverage is failing due to the formatting. For example: 
```ts
const envSettings = envSettingsParsed.success ? envSettingsParsed.data : {};
```
Since it is in one line, the else part of the ternary operator is considered covered even though its not executed. 

The formatting of the newly added pipeline settings creates a separate line, so its considered not covered:
```ts
const pipelineSettings = pipelineSettingsParsed.success
    ? pipelineSettingsParsed.data
    : {};
```

Should I implement a new test for this or is it ok that the coverage drops a bit?

jamacku reviewed (commented):
@engelmi Thank you for the PR. It looks great. Could you please also add the test into `tests/action.test.ts`?

You can take a Hardware test as an example:

https://github.com/sclorg/testing-farm-as-github-action/blob/080971e035d87c1bd99ef1235cb5a9f10ec8ae57/tests/action.test.ts#L255-L326

engelmi commented:
> @engelmi Thank you for the PR. It looks great. Could you please also add the test into `tests/action.test.ts`?
 
Hi @jamacku, 
sure thing. I added two tests - valid and invalid pipeline settings. I also switched from `safeParse` to `parse` so that an error in the schema doesn't get "covered up" and shows directly before submitting the TF request.

jamacku commented on tests/action.test.ts:
```diff
@@ -7,6 +7,7 @@ import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
 import action from '../src/action';
 import { NewRequest } from 'testing-farm';
 import { PullRequest } from '../src/pull-request';
+import { ZodError } from 'zod';
```
```suggestion
```

engelmi replied:
Done :heavy_check_mark:

jamacku reviewed (changes requested):
Thank you for adding tests. Please remove unused import in tests and please rebase to `main` and run `yarn && yarn run all`

jamacku reviewed (approved):
Thank you @engelmi, LGTM :+1:
